### PR TITLE
Only allow valid license identifiers in processed declared licenses

### PR DESCRIPTION
**Pull request by mnonnenmacher**
See the commit messages for details.

### CODE REVIEW

File: utils/src/main/kotlin/DeclaredLicenseProcessor.kt [hunk 1]
```diff
@@ -63,7 +63,7 @@ object DeclaredLicenseProcessor {
     fun process(declaredLicense: String): SpdxExpression? {
         val licenseWithoutPrefixOrSuffix = preprocess(declaredLicense)
```
Comment (fviernau) [inline utils/src/main/kotlin/DeclaredLicenseProcessor.kt:64]: commit-msg: Consider changing the last sentence to explain (shortly) in what way the reporting is actually improved. (As it is not obvious to me what it referrs to)

File: utils/src/test/kotlin/DeclaredLicenseProcessorTest.kt [hunk 2]
```diff
@@ -26,9 +26,11 @@ import org.ossreviewtoolkit.spdx.SpdxLicense
 import org.ossreviewtoolkit.spdx.SpdxLicenseAliasMapping
 import org.ossreviewtoolkit.spdx.SpdxLicenseIdExpression
 
+import io.kotest.core.spec.style.StringSpec
```
Comment (sschuberth) [inline utils/src/test/kotlin/DeclaredLicenseProcessorTest.kt:29]: Nit: While at it, you could also make this block go before the `org` imports.

File: utils/src/main/kotlin/DeclaredLicenseProcessor.kt (hunk 1 above)
Comment (mnonnenmacher) [reply utils/src/main/kotlin/DeclaredLicenseProcessor.kt:64]: Done.

File: utils/src/test/kotlin/DeclaredLicenseProcessorTest.kt (hunk 2 above)
Comment (mnonnenmacher) [reply utils/src/test/kotlin/DeclaredLicenseProcessorTest.kt:22]: Done.

**Pull request merged**